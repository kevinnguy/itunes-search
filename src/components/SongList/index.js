import React, { Component } from 'react';
import {
  FlatList,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

import Separator from '../Separator';
import SongRow from '../SongRow';

export default class SongList extends Component {
  keyExtractor = item => `${item.trackId}`;

  renderEmptyState = () => (
    <View style={styles.emptyState}>
      <Text style={styles.emptyStateText}>{'No search results'}</Text>
    </View>
  );

  renderSeparator = () => <Separator />;

  renderItem = ({ item }) => (
    <TouchableOpacity onPress={() => this.props.onPress(item)}>
      <SongRow data={item} />
    </TouchableOpacity>
  );

  render() {
    const { data } = this.props;
    return (
      <FlatList
        style={styles.container}
        data={data}
        ListEmptyComponent={this.renderEmptyState}
        ItemSeparatorComponent={this.renderSeparator}
        renderItem={this.renderItem}
        keyExtractor={this.keyExtractor}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    height: 200,
  },
  emptyState: {
    marginTop: 50,
    alignItems: 'center',
  },
  emptyStateText: {
    fontSize: 20,
    color: '#888',
  },
});
