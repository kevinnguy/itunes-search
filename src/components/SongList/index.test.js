import React from 'react';
import renderer from 'react-test-renderer';

import SongList from './index';
import mockData from '../../../__tests__/mockData.json';

test('renders correctly', () => {
  const tree = renderer.create(<SongList data={mockData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders empty state', () => {
  const tree = renderer.create(<SongList />).toJSON();
  expect(tree).toMatchSnapshot();
});
