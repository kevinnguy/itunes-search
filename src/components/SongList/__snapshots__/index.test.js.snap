// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  data={
    Object {
      "resultCount": 10,
      "results": Array [
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/better-together/1440857781?i=1440857786&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/94/25/9c/94259c23-84ee-129d-709c-577186cbe211/mzaf_5653537699505456197.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Better Together",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857786,
          "trackName": "Better Together",
          "trackNumber": 1,
          "trackPrice": 1.29,
          "trackTimeMillis": 207679,
          "trackViewUrl": "https://music.apple.com/us/album/better-together/1440857781?i=1440857786&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/be/38/d0/be38d058-31ed-c0ea-91e6-12052865fd20/source/100x100bb.jpg",
          "artworkUrl30": "https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/be/38/d0/be38d058-31ed-c0ea-91e6-12052865fd20/source/30x30bb.jpg",
          "artworkUrl60": "https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/be/38/d0/be38d058-31ed-c0ea-91e6-12052865fd20/source/60x60bb.jpg",
          "collectionCensoredName": "Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1469577723,
          "collectionName": "Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George",
          "collectionPrice": 9.99,
          "collectionViewUrl": "https://music.apple.com/us/album/upside-down/1469577723?i=1469577741&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview123/v4/75/5a/fc/755afca1-d911-be2b-c0fb-f99d6d52ce1a/mzaf_4156076990936187406.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2006-02-06T12:00:00Z",
          "trackCensoredName": "Upside Down",
          "trackCount": 14,
          "trackExplicitness": "notExplicit",
          "trackId": 1469577741,
          "trackName": "Upside Down",
          "trackNumber": 1,
          "trackPrice": 1.29,
          "trackTimeMillis": 208643,
          "trackViewUrl": "https://music.apple.com/us/album/upside-down/1469577723?i=1469577741&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/sitting-waiting-wishing/1440857781?i=1440857898&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/27/e6/28/27e628bb-17cb-eb94-87b4-d0f9b7ff1ebc/mzaf_20190415424710808.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Sitting, Waiting, Wishing",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857898,
          "trackName": "Sitting, Waiting, Wishing",
          "trackNumber": 6,
          "trackPrice": 1.29,
          "trackTimeMillis": 183721,
          "trackViewUrl": "https://music.apple.com/us/album/sitting-waiting-wishing/1440857781?i=1440857898&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/banana-pancakes/1440857781?i=1440857795&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/51/ba/f8/51baf8ca-e417-56e8-d6e1-4f02634e451c/mzaf_5766837215935557210.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Banana Pancakes",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857795,
          "trackName": "Banana Pancakes",
          "trackNumber": 3,
          "trackPrice": 1.29,
          "trackTimeMillis": 191854,
          "trackViewUrl": "https://music.apple.com/us/album/banana-pancakes/1440857781?i=1440857795&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/good-people/1440857781?i=1440857797&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview128/v4/d2/bf/ba/d2bfba21-a41e-c4f0-cfb7-c83420a77336/mzaf_3810532136343461295.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Good People",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857797,
          "trackName": "Good People",
          "trackNumber": 4,
          "trackPrice": 1.29,
          "trackTimeMillis": 208509,
          "trackViewUrl": "https://music.apple.com/us/album/good-people/1440857781?i=1440857797&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/breakdown/1440857781?i=1440857908&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview128/v4/85/ae/a8/85aea863-ed49-6ab9-bf58-cee20cd312fc/mzaf_4839543551345827301.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Breakdown",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857908,
          "trackName": "Breakdown",
          "trackNumber": 11,
          "trackPrice": 1.29,
          "trackTimeMillis": 212986,
          "trackViewUrl": "https://music.apple.com/us/album/breakdown/1440857781?i=1440857908&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/42/24/53/422453e1-9a78-34e6-6491-3c54469c28ec/source/100x100bb.jpg",
          "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/42/24/53/422453e1-9a78-34e6-6491-3c54469c28ec/source/30x30bb.jpg",
          "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/42/24/53/422453e1-9a78-34e6-6491-3c54469c28ec/source/60x60bb.jpg",
          "collectionCensoredName": "To the Sea (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440851963,
          "collectionName": "To the Sea (Bonus Track Version)",
          "collectionPrice": 9.99,
          "collectionViewUrl": "https://music.apple.com/us/album/you-and-your-heart/1440851963?i=1440851968&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview128/v4/16/10/ff/1610fffd-afb3-c914-ab3c-b58ba93d06b1/mzaf_356097296229876349.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2010-04-06T12:00:00Z",
          "trackCensoredName": "You and Your Heart",
          "trackCount": 14,
          "trackExplicitness": "notExplicit",
          "trackId": 1440851968,
          "trackName": "You and Your Heart",
          "trackNumber": 1,
          "trackPrice": 1.29,
          "trackTimeMillis": 192853,
          "trackViewUrl": "https://music.apple.com/us/album/you-and-your-heart/1440851963?i=1440851968&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/100x100bb.jpg",
          "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/30x30bb.jpg",
          "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/60x60bb.jpg",
          "collectionCensoredName": "Sleep Through the Static",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440854851,
          "collectionName": "Sleep Through the Static",
          "collectionPrice": 13.99,
          "collectionViewUrl": "https://music.apple.com/us/album/if-i-had-eyes/1440854851?i=1440855373&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/bf/d2/ad/bfd2ad0d-0e1e-20d1-0a73-898c1b1badcf/mzaf_5135296513885819410.plus.aac.p.m4a",
          "primaryGenreName": "Alternative",
          "releaseDate": "2007-12-15T12:00:00Z",
          "trackCensoredName": "If I Had Eyes",
          "trackCount": 14,
          "trackExplicitness": "notExplicit",
          "trackId": 1440855373,
          "trackName": "If I Had Eyes",
          "trackNumber": 6,
          "trackPrice": 1.29,
          "trackTimeMillis": 238806,
          "trackViewUrl": "https://music.apple.com/us/album/if-i-had-eyes/1440854851?i=1440855373&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/100x100bb.jpg",
          "artworkUrl30": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/30x30bb.jpg",
          "artworkUrl60": "https://is5-ssl.mzstatic.com/image/thumb/Music118/v4/49/09/ff/4909ffd6-9051-c729-7761-3dfcc183333c/source/60x60bb.jpg",
          "collectionCensoredName": "Sleep Through the Static",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440854851,
          "collectionName": "Sleep Through the Static",
          "collectionPrice": 13.99,
          "collectionViewUrl": "https://music.apple.com/us/album/angel/1440854851?i=1440855248&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/57/93/94/57939483-f16a-3fee-dd39-de7df73e8413/mzaf_5320784781866992253.plus.aac.p.m4a",
          "primaryGenreName": "Alternative",
          "releaseDate": "2008-02-02T12:00:00Z",
          "trackCensoredName": "Angel",
          "trackCount": 14,
          "trackExplicitness": "notExplicit",
          "trackId": 1440855248,
          "trackName": "Angel",
          "trackNumber": 4,
          "trackPrice": 1.29,
          "trackTimeMillis": 122694,
          "trackViewUrl": "https://music.apple.com/us/album/angel/1440854851?i=1440855248&uo=4",
          "wrapperType": "track",
        },
        Object {
          "artistId": 909253,
          "artistName": "Jack Johnson",
          "artistViewUrl": "https://music.apple.com/us/artist/jack-johnson/909253?uo=4",
          "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
          "artworkUrl30": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/30x30bb.jpg",
          "artworkUrl60": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/60x60bb.jpg",
          "collectionCensoredName": "In Between Dreams (Bonus Track Version)",
          "collectionExplicitness": "notExplicit",
          "collectionId": 1440857781,
          "collectionName": "In Between Dreams (Bonus Track Version)",
          "collectionPrice": 11.99,
          "collectionViewUrl": "https://music.apple.com/us/album/do-you-remember/1440857781?i=1440857917&uo=4",
          "country": "USA",
          "currency": "USD",
          "discCount": 1,
          "discNumber": 1,
          "isStreamable": true,
          "kind": "song",
          "previewUrl": "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview118/v4/da/c5/2d/dac52de0-55e9-8585-f91c-f92ffc519a59/mzaf_7471265503660630513.plus.aac.p.m4a",
          "primaryGenreName": "Rock",
          "releaseDate": "2005-03-01T12:00:00Z",
          "trackCensoredName": "Do You Remember",
          "trackCount": 15,
          "trackExplicitness": "notExplicit",
          "trackId": 1440857917,
          "trackName": "Do You Remember",
          "trackNumber": 13,
          "trackPrice": 1.29,
          "trackTimeMillis": 144006,
          "trackViewUrl": "https://music.apple.com/us/album/do-you-remember/1440857781?i=1440857917&uo=4",
          "wrapperType": "track",
        },
      ],
    }
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#888",
            "fontSize": 20,
          }
        }
      >
        No search results
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders empty state 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#888",
            "fontSize": 20,
          }
        }
      >
        No search results
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
